feat(expr): add area as an alias for st_area

tisonkun opened the pull request:
### Task Description

This closes https://github.com/oceanbase/oceanbase/issues/1398.

### Solution Description

Since OB already implements ST_Area, we add Area as an alias as MySQL definitions.

### Passed Regressions

TRIVIAL

### Upgrade Compatibility

N/A

### Other Information

N/A

### Release Note

Add Area as an alias for ST_Area.

hnwyllmm commented:
I'm so sorry that it takes so long time to response to you. We had published a mysqltest of geometry. Could you add a mysqltest in the directory 'tools/deploy/mysql_test/test_suite/geometry/' about area?

tisonkun commented:
@hnwyllmm Thanks for your updates! I'll investigate it in days.

tisonkun commented:
How can I run the tests locally? Otherwise, it's extremely a challenge to read the text and debug in mind.

tisonkun commented:
I'm rebasing this PR and trying to run the follow queries against my local OB server to get a result...

```sql
select x, y, st_astext(poi) from gis_point; -- existing

-- added
select x, y, area(poi) from gis_point;
select x, y, st_area(poi) from gis_point;
```

tisonkun commented:
Encounter `Spatial reference system is empty`. What should I configure?>

tisonkun commented:
Pushed a commit to add the test cases. Let me see how to generate the expected result.

Currently, running on a self-compiled cluster encounter the "Spatial reference system is empty" issue, running on a MySQL cluster encounter:

```
mysql> create table if not exists gis_point (
    ->         x decimal(19,16),
    ->         y decimal(19,16),
    ->         poi POINT not null srid 4326,
    ->         index `xy_index` (x, y) local,
    ->         index `poi_index` (poi) local
    ->         );
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'local,
        index `poi_index` (poi) local
        )' at line 5
```

I'll try if I can run against the OB docker image since this only adds an alias instead of changing the behavior.

tisonkun commented:
Updated.

I found one issue here - `area` is deprecated in 5.7 and removed in 8.0.

So, depending on which version you'd like to be compatible with, this alias may not need to be added. But the test coverage is still valuable, whlie I can only add a very simple case now.

hnwyllmm commented:
> Hi @hnwyllmm! May I ask how does OB merge a PR? My scenario is that I may merge master to catch up some nightly changes, but if you don't use the "Squash and merge" approach, it may cause a merge commit in the master branch after merging.

We use "rebase" or "squash" mode to merge the code, so I think it's OK.

hnwyllmm commented:
> How can I run the tests locally? Otherwise, it's extremely a challenge to read the text and debug in mind.

Please use the script to run the mysql test and the file is :
tools/deploy/obd.sh
you can type obd.sh -h to see how to use.

hnwyllmm commented:
> Updated.
> 
> I found one issue here - `area` is deprecated in 5.7 and removed in 8.0.
> 
> So, depending on which version you'd like to be compatible with, this alias may not need to be added. But the test coverage is still valuable, whlie I can only add a very simple case now.

I think it's valuable as the MySQL 5.7 is used by so many companies and it will take a long time to upgrade even though the latest edition is 8.0.

hnwyllmm commented:
And thanks a lot for the new mysql test. But could you write the area test code to a new file? such as geometry_area.test?

tisonkun commented:
@hnwyllmm Done.

Although I don't know the prologue of `geometry_basic_mysql.test` well, I copied it to the new file since I guess it does some initialization works.

tisonkun commented:
@hnwyllmm may I ask what is the status of this PR now? Is there any blocker or I should update?

hnwyllmm commented:
It works all right and we are running more tests now.
